api/capi/src/nnstreamer-capi-single-new.c

```diff
@@ -523,10 +533,13 @@ ml_single_close (ml_single_h single)
     return ML_ERROR_INVALID_PARAMETER;
   }
 
+  g_mutex_lock (&ml_single_mutex);
   single_h = (ml_single *) single;
   ML_SINGLE_MAGIC_CHECK (single_h);
 
   single_h->magic = 0;
+  g_mutex_unlock (&ml_single_mutex);
+
```

Comment:
Don't we need to protect ```single_h->magic``` access with single_h->mutex (what about moving LINE 543 -->  LINE 539)? So that other thread in a critical section of single_h->mutex won't get its magic value reset.?

Reply:
This is resolved with global lock